lambda_src/lambda_function.py

```diff
@@ -33,19 +38,22 @@ def async_flow_init(event: Any, context: Any) -> Dict[Text, Any]:
     LOG.debug('Found a destination header and hence using async_flow_init().')
 
     headers = event['headers']
+    req_body = loads(event['body'])
     batch_id = headers[BATCH_ID_HEADER]
     destination = headers[DESTINATION_URI_HEADER]
-    headers.pop(DESTINATION_URI_HEADER)
```

Comment:
I don't think this breaks anything, but it also doesn't seem to do anything helpful. if you want to refactor this, let's maybe do something like 

```python
destination = headers['write-uri'] = headers.pop(DESTINATION_URI_HEADER)
```

as that simplifies the flow a little bit?